kubernetes/kubernetes.go

```diff
@@ -7,20 +7,57 @@ import (
 	
 	kube "k8s.io/client-go/1.5/kubernetes"
 	"k8s.io/client-go/1.5/rest"
+	"k8s.io/client-go/1.5/tools/clientcmd"
 )
 
+// Check if running in a cluster with service accounts
+// Use case: firewall based/only cluster 
+func RunningInCluster() (bool)  {
+        if _, err := os.Stat("/var/run/secrets/kubernetes.io/serviceaccount/token"); err == nil {
+                return true;
+        } else {
+                return false;
+        }
+}
+
+func GetConfig() (*rest.Config, error) {
+        var kubeConfig rest.Config
+
+        // Set the Kubernetes configuration based on the environment
+        if RunningInCluster() {
+                config, err := rest.InClusterConfig()
+
+                if err != nil {
+                        return nil, fmt.Errorf("Failed to create in-cluster config: %v.", err)
+                }
+
+                kubeConfig = *config
+        } else {
+                loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+                configOverrides := &clientcmd.ConfigOverrides{}
+                config := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+                tmpKubeConfig, err := config.ClientConfig()
+                if err != nil {
+                        return nil, fmt.Errorf("Failed to load local kube config: %v", err)
+                }
+                kubeConfig = *tmpKubeConfig;
+        }
+
+	if apiserverHost, override := cfg.ClusterAPIServerHost(); override {
+		fmt.Printf("API server host overriden to: %s\n", apiserverHost)
```

Comment:
This file is still using `fmt` (pre-glog commit) but is no longer importing the fmt package.

Here are the errors I'm seeing:

```
GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o kube-monkey
# github.com/asobti/kube-monkey/kubernetes
kubernetes/kubernetes.go:16:22: undefined: os
kubernetes/kubernetes.go:31:37: undefined: fmt
kubernetes/kubernetes.go:41:37: undefined: fmt
kubernetes/kubernetes.go:47:3: undefined: fmt
kubernetes/kubernetes.go:48:3: undefined: config
```

Reply:
Ahh yes, I was on the go and unable to integrate the glog changes. It should now be using it glog style.